### DISCUSSION TITLE
User language detection

### DISCUSSION THREAD

**Discussion by stonebuzz** [squidfunk/mkdocs-material#6052] [Q&A] [answered]
I was wondering if there was a way to implement dynamic location detection, so that if a user is,in france, the plugin will detect their region and redirect the user to ```https://my_doc/fr/``` (and a fallback to ```/en/``` if desired language is not available).

I use the ```mkdocs-static-i18n``` plugin, but it doesn't allow automatic detection. 

Best regards

**Comment by alexvoss** [comment, marked as answer]
I could not find a mkdocs plugin for this. It certainly should be possible to do this using some JavaScript, perhaps to forward to a language version absent a user choice? I would hesitate to recommend an approach but definitely look at [navigator.languages](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages). StackOverflow may be your friend there.

**Reply by stonebuzz** [reply]
Thanks for your reply

Best regards

**Reply by alexvoss** [reply]
You're welcome. Feel free to ask follow-on questions if necessary. I imagine that developing a generic version might be a bit of work and there might be question arising but perhaps what I have pointed you at is sufficient for your use case.

**Reply by ztheory** [reply]
Using the user-agent data to detect the language of the browser is the way to go, rather than IP geolocation.

Example user agent field/value in the request headers:
`Accept-Language: en-US,en;q=0.5`
